src/NUnitTestExtractor/NUnitTestExtractorApp.cs

```diff
@@ -16,383 +15,264 @@ public static class NUnitTestExtractorApp
                 if (parser.ParseArgumentsStrict(args, _options, () => Environment.Exit(-1)))
                 {
-                    if (_options.Help)
-                    {
-                        Console.WriteLine(CommandLine.Text.HelpText.AutoBuild(_options));
-                        return;
-                    }
-                   
-                    if(!string.IsNullOrEmpty(_options.Output))
+                    try
                     {
-                        string directory = null;
-
-                        int index = _options.Output.LastIndexOf("/", StringComparison.OrdinalIgnoreCase);
-
-                        if (index > 0)
+                        if (_options.Help)
                         {
-                            directory = _options.Output.Substring(0, index);
-                        }  
+                            Console.WriteLine(CommandLine.Text.HelpText.AutoBuild(_options));
 
-                        if (directory!=null && !Directory.Exists(directory))
-                        {
-                            Directory.CreateDirectory(directory);
+                            Environment.Exit(0);
                         }
 
-                        if (!File.Exists(_options.Output))
+                        // _options.Output is an abusolute path of output file for the test name list
+                        if (!string.IsNullOrEmpty(_options.Output))
                         {
-                            using (FileStream stream = File.Create(_options.Output)) { }
+                            CreateOutputFile();
                         }
-                    }
 
-                    if (_options.Level != null)
-                    {
-                        StreamWriter writer = null;
-                        level = ParseLevel(_options.Level);
-                        try
-                        {
-                            if (_options.Output == null)
-                            {
-                                writer = new StreamWriter(Console.OpenStandardOutput());
-                            }
-                            else
-                            {
-                                writer = new StreamWriter(_options.Output, append: true);
-                            }
-
-                            GetTests(_options.DLLs, level, writer);
-                        }
-                        finally
+                        foreach (var assembly in _options.DLLs)
                         {
-                            if (writer != null)
-                            {
-                                writer.Dispose();
-                                
-                            }
+                            GenerateInfoFromAssembly(assembly);
                         }
                     }
+                    catch (Exception e)
+                    {
+                        Console.WriteLine(StringUtils.FormatInvariant("Failed to process: {0}", e.InnerException));
```

Comment:
[Must] Write to stderr instead of stdout.

Reply:
Fixed